### PR TITLE
add WakeupEvents support to perf Reader

### PR DESCRIPTION
**Pull request by brycekahle**
follow up to #524 with only the `WakeupEvents` portion.

### CODE REVIEW

File: perf/ring.go [hunk 1]
```diff
@@ -98,13 +98,20 @@ func (ring *perfEventRing) Close() {
 	ring.mmap = nil
 }
 
-func createPerfEvent(cpu, watermark int, overwritable bool) (int, error) {
-	if watermark == 0 {
-		watermark = 1
+func createPerfEvent(cpu int, opts ReaderOptions) (int, error) {
+	wakeup := 0
+	bits := 0
```
Comment (florianl) [inline perf/ring.go:103]: As the documentation for `WakeupEvents` states, it should be mutually exclusive with `Watermark`.
Can we add a check for the case where both, `WakeupEvents` and `Watermark`, are set  and return something like `fmt.Errorf("WakeupEvents and Watermark are mutually exclusive: %w", unix.EINVAL)`.

Comment (brycekahle) [reply perf/ring.go:103]: Like this one? https://github.com/cilium/ebpf/pull/1404/files#diff-7b72e423487d9bde0d352acf2654d5e70f0c10898776b867fc3bac2b38f9ceafR199-R201

Comment (florianl) [reply perf/ring.go:103]: ah - good point! placing the check in `NewReaderWithOptions()` before doing allocations make sense!

**Review by florianl — APPROVED** [review]

**Comment by dylandreimerink** [comment]
Hi! It seems this PR introduced a flaky test, see https://github.com/cilium/ebpf/issues/1419. @brycekahle could you take a look? So far I have not been able to find the root cause, will update the issue if I do.

**Pull request merged**